pages/dev/general-message-passing/cosmos-gmp.mdx

```diff
@@ -0,0 +1,165 @@
+import { Callout } from 'nextra-theme-docs'
+
+
+# Cosmos General Message Passing
+
+Axelar has expanded General Message Passing (GMP) to support Cosmos blockchains. With Axelar, you can now send and receive messages on EVM chains and Cosmos chains. Messages sent to Cosmos chains can be received by [CosmWasm smart contracts](https://github.com/axelarnetwork/evm-cosmos-gmp-sample/tree/main/cosmwasm-integration/) (on blockchains with CosmWasm support), or those messages can be received [natively at the consensus layer as part Go code](https://github.com/axelarnetwork/evm-cosmos-gmp-sample/tree/main/native-integration).
+
+
+<Callout emoji="💡">
+To see full examples of sending and receiving messages in all cases, check out our full [Cosmos GMP Example Project](https://github.com/axelarnetwork/evm-cosmos-gmp-sample/).
+</Callout>
+
+## Messages from EVM to CosmWasm
+To send a message from an EVM chain to a Cosmos CosmWasm contract, the payload must be encoded before calling the [standard](./gmp-messages) `payNativeGasForContractCall` and `callContract` methods (or [token-including equivalents](./gmp-tokens-with-messages)). You must also specify the destination method you are calling.
+
+
+The payload must specify:
+
+1. **string** - The method name of the CosmWasm contract to call
+1. **string[]** - The names of all arguments
+1. **string[]** - the types of all arguments
+1. **bytes** - the values of all arguments
+
+### MultiSend Example encoding
+This example encodes a string array of recipients, as you can see in our [example of sending tokens and recipients](https://github.com/axelarnetwork/axelar-examples/blob/main/examples/evm/call-contract-with-token/DistributionExecutable.sol).
+
+```solidity
+bytes memory argValue = abi.encode(recipients); // A standard EVM payload
+bytes memory payload  = abi.encode(
+    "multi_send", // CosmWasm method name
+    StringArray.fromArray1(["recipients"]), // argument name
+    StringArray.fromArray1(["string[]"]), // argument type
+    argValue // argument value
+);
+bytes memory payloadToCW = abi.encodePacked(
+    bytes4(uint32(1)), // version number
+    payload
+);
+```
```

Comment:
There is an another way to call a CosmWasm contract https://github.com/axelarnetwork/evm-cosmos-gmp-sample/tree/main/cosmwasm-integration#1-pass-json-cosmwasm-contract-call-directly